sdk/core/azure-core/src/main/java/com/azure/core/implementation/jackson/HeaderCollectionHandler.java

```diff
@@ -15,15 +15,18 @@
 import java.util.HashMap;
 import java.util.Locale;
 import java.util.Map;
-import java.util.Optional;
 import java.util.concurrent.ConcurrentHashMap;
 
 /*
  * Internal helper class that helps manage converting headers into their header collection.
  */
 final class HeaderCollectionHandler {
     private static final int CACHE_SIZE_LIMIT = 10000;
-    private static final Map<Field, Optional<MethodHandle>> FIELD_TO_SETTER_CACHE = new ConcurrentHashMap<>();
+    private static final Map<Field, MethodHandle> FIELD_TO_SETTER_CACHE = new ConcurrentHashMap<>();
+
+    // Dummy constant that indicates no setter was found for the Field.
+    private static final MethodHandle NO_SETTER_HANDLE = MethodHandles.identity(HeaderCollectionHandler.class);
```

Comment:
I might be paranoic here, but would it make sense to create a `private static class NullSetterHandle` just in case ?

Reply:
Are you asking for `private static class NullSetterHandle extends MethodHandle`? This won't be possible as `MethodHandle`'s constructor is package-private.

Reply:
No. `private static final MethodHandle NO_SETTER_HANDLE = MethodHandles.identity(ImaginaryPrivateNotRelevantInnerThatNeverGetsTouchedEver.class);`

Reply:
I'm wondering if it'd be better to just use a primitive here like `int.class` or `Object.class`?

Reply:
we won't be attempting to resolve setters for `HeaderCollectionHandler` ? Let's leave it.